src/Utility/Hash.php

```diff
@@ -415,6 +420,10 @@ public static function remove(ArrayAccess|array $data, string $path): ArrayAcces
 
         [$token, $conditions] = self::_splitConditions($token);
 
+        if (!is_iterable($data)) {
+            throw new CakeException('Cannot use path tokens of type `{}` or `[]` for non-iterable objects.');
+        }
```

Comment:
Ok, this is why I closed the PR originally, I wasn't sure if we had to add ArrayAccess-only iteration and logic.